File: src/main/java/com/turchinsky/entities/TrainEntity.java
package com.turchinsky.entities;


import javax.persistence.*;

@Entity
@Table(name = "train")
public class TrainEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Basic
    @Column
    private String number;

    @Basic
    @Column
    private String type;

    @Basic
    @Column
    private int capacity;

    @Basic
    @Column
    private int velocity;

//    @OneToMany(mappedBy = "refTrainEntity")
//    private List<ScheduleDetailsEntity> refScheduleDetailsEntities;
//
//    @OneToMany(mappedBy = "refTrainEntity")
//    private List<TrainHasScheduleAndRouteEntity> refTrainHasScheduleAndRouteEntities;
//

    public void setId(int id) {
        this.id = id;
    }


    public int getId() {
        return id;
    }

    public String getNumber() {
        return number;
    }

    public void setNumber(String number) {
        this.number = number;
    }

    public String getType() {
        return type;
    }

    public void setType(String name) {
        this.type = name;
    }

    public int getCapacity() {
        return capacity;
    }

    public void setCapacity(int capacity) {
        this.capacity = capacity;
    }


    public int getVelocity() {
        return velocity;
    }

    public void setVelocity(int velocity) {
        this.velocity = velocity;
    }

//    public List<ScheduleDetailsEntity> getRefScheduleDetailsEntities() {
//        return refScheduleDetailsEntities;
//    }
//
//    public void setRefScheduleDetailsEntities(
//            List<ScheduleDetailsEntity> refScheduleDetailsEntities) {
//        this.refScheduleDetailsEntities = refScheduleDetailsEntities;
//    }
//
//    public List<TrainHasScheduleAndRouteEntity> getRefTrainHasScheduleAndRouteEntities() {
//        return refTrainHasScheduleAndRouteEntities;
//    }
//
//    public void setRefTrainHasScheduleAndRouteEntities(
//            List<TrainHasScheduleAndRouteEntity> refTrainHasScheduleAndRouteEntities) {
//        this.refTrainHasScheduleAndRouteEntities = refTrainHasScheduleAndRouteEntities;
//    }

    @Override
    public String toString() {
        return "TrainEntity{" +
                "id=" + id +
                ", number='" + number + '\'' +
                ", type='" + type + '\'' +
                ", capacity=" + capacity +
                ", velocity=" + velocity +
                '}';
    }

}

File: src/main/java/com/turchinsky/kt/TrainEntity.kt
//package com.turchinsky.kt
//
//import javax.persistence.*
//
//@Entity
//@Table(name = "train", schema = "public", catalog = "sbb")
//open class TrainEntity {
//    @get:Id
//    @get:Column(name = "id", nullable = false, insertable = false, updatable = false)
//    var id: Int? = null
//
//    @get:Basic
//    @get:Column(name = "capacity", nullable = true)
//    var capacity: Int? = null
//
//    @get:Basic
//    @get:Column(name = "type", nullable = true)
//    var type: String? = null
//
//    @get:Basic
//    @get:Column(name = "number", nullable = true)
//    var number: String? = null
//
//    @get:Basic
//    @get:Column(name = "velocity", nullable = true)
//    var velocity: Int? = null
//
//    @get:OneToMany(mappedBy = "refTrainEntity")
//    var refScheduleDetailsEntities: List<ScheduleDetailsEntity>? = null
//
//    @get:OneToMany(mappedBy = "refTrainEntity")
//    var refTrainHasScheduleAndRouteEntities: List<TrainHasScheduleAndRouteEntity>? = null
//
//    override fun toString(): String =
//            "Entity of type: ${javaClass.name} ( " +
//                    "id = $id " +
//                    "capacity = $capacity " +
//                    "type = $type " +
//                    "number = $number " +
//                    "velocity = $velocity " +
//                    ")"
//
//    // constant value returned to avoid entity inequality to itself before and after it's update/merge
//    override fun hashCode(): Int = 42
//
//    override fun equals(other: Any?): Boolean {
//        if (this === other) return true
//        if (javaClass != other?.javaClass) return false
//        other as TrainEntity
//
//        if (id != other.id) return false
//        if (capacity != other.capacity) return false
//        if (type != other.type) return false
//        if (number != other.number) return false
//        if (velocity != other.velocity) return false
//
//        return true
//    }
//
//}
//
